src/vmm/src/resources.rs

```diff
@@ -111,6 +116,10 @@ impl VmResources {
         if let Some(vsock_config) = vmm_config.vsock_device {
             resources.set_vsock_device(vsock_config);
         }
+
+        if let Some(mmds_config) = vmm_config.mmds_config {
+            resources.set_mmds_config(mmds_config);
+        }
```

Comment:
Can you also test this in `test_from_json()` by creating a new json with, for example, an invalid `ipv4_address` value and checking the returned error?